example/calculate_pi/calculate_pi.hpp

```diff
@@ -18,25 +18,27 @@
 #include <vector>
 
 // Parthenon Includes
+#include <interface/state_descriptor.hpp>
 #include <parthenon/package.hpp>
 
 namespace calculate_pi {
 using namespace parthenon::package::prelude;
+using parthenon::Packages_t;
+using parthenon::ParArrayHost;
+using Pack_t = parthenon::MeshBlockVarPack<Real>;
 
 // Package Callbacks
 void SetInOrOut(std::shared_ptr<Container<Real>> &rc);
 std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin);
 
 // Task Implementations
-// task per meshblock
-parthenon::TaskStatus ComputeArea(parthenon::MeshBlock *pmb);
-// Task over whole mesh, no packs
-parthenon::TaskStatus RetrieveAreas(std::vector<parthenon::MeshBlock *> &blocks,
-                                    parthenon::Packages_t &packages);
-
-// Run task on the entire mesh at once
-parthenon::TaskStatus ComputeAreaOnMesh(std::vector<parthenon::MeshBlock *> &blocks,
-                                        parthenon::Packages_t &packages);
+// Note pass by value here. Required for capture.
+// All objects here have reference semantics, so capture by value is ok.
+// TODO(JMM) A std::shared_ptr might be better.
```

Comment:
Why would `shared_ptr` be better? Aren't the heavy bits of `Pack_t` and `ParArrayHost` kokkos views, which are essentially the same as `shared_ptr`?

Reply:
Yeah, that's why I didn't implement a `shared_ptr`. I was just concerned about metadata. If you carry enough metadata around with your pointer to your big heavy array, you'll eventually start to see a cost.

My thinking was don't add a `shared_ptr` now but keep it in mind in case it becomes an issue.